packages/augur-core/source/contracts/trading/ZeroXTrade.sol

```diff
@@ -308,9 +317,14 @@ contract ZeroXTrade is Initializable, IZeroXTrade, IERC1155 {
             address(this),
             _tokenIds,
             _tokenValues,
-            _callbackData,
-            _kycToken
+            _callbackData
         );
+
+        uint256 _assetDataLength = _assetData.length;
+
+        // We must pad the kycToken as the 0x exchnage requires the assetData be a multiple of 32 not counting the selector
+        _assetData = abi.encodePacked(_assetData, bytes32(uint256(address(_kycToken))));
```

Comment:
Won't `abi.encode(_assetData, _kycToken)` achieve this?  Or will it do other things to the `_assetData` that are undesirable?

Reply:
The comment above the line explains. A simple `abi.encode` will not have the correct length and will get rejected by 0x.

Reply:
Ah, it is because `abi.encode` will offset the dynamic `_assetData`:
> Non-standard Packed Mode [...] dynamic types are encoded in-place and without the length.